src/authextension.ts

```diff
@@ -15,26 +17,65 @@ export module authextension {
     process.env['THREE_SCALE_USER_TOKEN'] = context_f8_3scale_user_key;
   };
 
+  setUUID = (context_UUID) => {
+    process.env['UUID'] = context_UUID;
+  };
+
   export const authorize_f8_analytics = context => {
     return new Promise((resolve, reject) => {
+      let respData;
       let context_f8_access_routes = context.globalState.get(
         'f8_access_routes'
       );
       let context_f8_3scale_user_key = context.globalState.get(
         'f8_3scale_user_key'
       );
+
       if (context_f8_access_routes && context_f8_3scale_user_key) {
         setContextData(context_f8_access_routes, context_f8_3scale_user_key);
-        resolve(true);
       } else {
-        get_3scale_routes(context)
-          .then(resp => {
-            resolve(resp);
-          })
-          .catch(err => {
-            reject(err);
-          });
+        respData = get_3scale_routes(context);
+        if (!respData) {
+          reject(respData);
+        }
+      }
+
+      let context_UUID = context.globalState.get(GlobalState.UUID);
+
+      if (context_UUID && context_UUID != '') {
+        setUUID(context_UUID);
+      } else {
+        respData = get_UUID(context);
+        if (!respData) {
+          reject(respData);
+        }
       }
+
+      resolve(true);
+    });
+  };
+
+  export const get_UUID = context => {
+    return new Promise((resolve, reject) => {
+      let options = {};
+      options['uri'] = `${
+        Apiendpoint.OSIO_ROUTE_URL
+        }/user?user_key=${Apiendpoint.THREE_SCALE_CONNECT_KEY}`;
+
+      stackAnalysisServices
+        .getUUIDService(options)
+        .then(respData => {
+          let uuid = respData;
+          if (uuid) {
+            context.globalState.update(GlobalState.UUID, uuid);
+            let context_UUID = context.globalState.get(GlobalState.UUID);
+            setUUID(context_UUID);
+            resolve(true);
+          }
+        })
+        .catch(err => {
+          reject(null);
+        });
```

Comment:
Use async/await pairs instead promise callbacks, it is improves the code readability by a lot.
Also get rid the usage of requests. npm requests package has been deprecated and shouldn't be used. Either use fetch or axios.

```suggestion
export async function getUUID() {
      const url = `${
        Apiendpoint.OSIO_ROUTE_URL
        }/user?user_key=${Apiendpoint.THREE_SCALE_CONNECT_KEY}`;
        return await fetch(url, {method: 'POST'});
 }
```

Reply:
Ack, Updated the code: using fetch and async/await pairs.